Tribler/Test/Core/Modules/RestApi/test_events_endpoint.py

```diff
@@ -0,0 +1,62 @@
+import json
+from twisted.internet import reactor
+from twisted.internet.defer import Deferred
+from twisted.internet.protocol import Protocol
+from twisted.web.client import Agent
+from twisted.web.http_headers import Headers
+from Tribler.Core.Utilities.twisted_thread import deferred
+from Tribler.Core.simpledefs import SIGNAL_CHANNEL, SIGNAL_ON_SEARCH_RESULTS, SIGNAL_TORRENT
+from Tribler.Core.version import version_id
+from Tribler.Test.Core.Modules.RestApi.base_api_test import AbstractApiTest
+
+
+class EventDataProtocol(Protocol):
+    """
+    This class is responsible for reading the data received over the event socket.
+    """
+    def __init__(self, messages_to_wait_for, finished):
+        self.json_buffer = []
+        self.messages_to_wait_for = messages_to_wait_for + 1
```

Comment:
That's due to the fact that the first event message is always `events_start` so the events we are looking for are coming after that one. This was indeed not clear so I've added a comment.